com.unity.perception/Tests/Runtime/Randomization/ScenarioTests/ScenarioTests.cs

```diff
@@ -75,11 +75,11 @@ public void ScenarioConfigurationOverwrittenDuringDeserialization()
 
             // Serialize some values
             m_Scenario.constants = constants;
-            var serializedConfig = m_Scenario.SerializeToJson();
+            m_Scenario.configuration = new TextAsset(m_Scenario.SerializeToJson());
 
             // Change the values
             m_Scenario.constants = changedConstants;
-            m_Scenario.DeserializeFromJson(serializedConfig);
+            m_Scenario.DeserializeConfiguration();
```

Comment:
We don't necessarily want to de-serialize the config at will, we just want to differentiate between config assignment and config de-serialization so an editor assigned config can be deserialized without "reassigning" it when the scenario awakes.

With this in mind, the changes I'll make include:
- Making the DeserializeConfiguration() method protected so it can still be overridden by the UnitySimulationScenario class while removing it from the public interface
- Adding a DeserializeConfigurationInternal() method so we can continue to de-serialize configs for editor and testing purposes